### PR TITLE
AuthProxy: replaces setting ldap_sync_ttl with sync_ttl

### LINKED ISSUE #18586: Incorrect LDAP option for the auth proxy
**What happened**:
According to [documentation](https://grafana.com/docs/auth/auth-proxy/) option `ldap_sync_ttl` should be used only with LDAP integration

**What you expected to happen**:
This option is used even if there LDAP integration is not enabled

**Environment**:
- Grafana version: 6.4-pre

### PR DESCRIPTION
**Pull request by jongyllen**
**What this PR does / why we need it**:
It adds a new config value, `sync_ttl` under `[auth.proxy]` since the existing `ldap_sync_ttl` affects the ttl regardless if ldap is used or not, making it misguiding.

**Which issue(s) this PR fixes**:
Fixes #18586 

**Special notes for your reviewer**:
Keeping the old `ldap_sync_ttl` around for compatibility reasons and environment variables.

We also discussed having default value for `ldap_sync_ttl` as either -1 or 60 to know if it was overridden in custom.ini. I ended up choosing 60 as it might make more sense

### CODE REVIEW

File: pkg/setting/setting.go [hunk 1]
```diff
@@ -854,7 +854,17 @@ func (cfg *Cfg) Load(args *CommandLineArgs) error {
 		return err
 	}
 	AuthProxyAutoSignUp = authProxy.Key("auto_sign_up").MustBool(true)
-	AuthProxyLDAPSyncTtl = authProxy.Key("ldap_sync_ttl").MustInt()
+
+	ldapSyncVal := authProxy.Key("ldap_sync_ttl").MustInt()
```
Comment (torkelo) [inline pkg/setting/setting.go]: you can pass in default (60) to the MustInt function. and maybe define that as a constant? AUTH_PROXY_SYNC_TTL (still needs to match the one defined in defaults.ini though.. so add that as a comment on the constant)

**Review by torkelo — APPROVED** [review]
Looks good! just one minor comment

**Review by xlson — APPROVED** [review]
Looks good! Minor nitpicking regarding test names.

File: pkg/setting/setting_test.go [hunk 2]
```diff
@@ -227,6 +227,50 @@ func TestLoadingSettings(t *testing.T) {
...
+			err := cfg.Load(&CommandLineArgs{
+				HomePath: "../../",
+				Args:     []string{"cfg:auth.proxy.ldap_sync_ttl=5"},
+			})
+			So(err, ShouldBeNil)
+
+			So(AuthProxySyncTtl, ShouldEqual, 5)
+		})
+
+		Convey("ldap_sync should override ldap_sync_ttl that is -1", func() {
```
[the remark this answers is not in the document]
Comment (jongyllen) [reply pkg/setting/setting_test.go]: Thanks, fixed

File: pkg/setting/setting_test.go [hunk 3]
```diff
@@ -227,6 +227,50 @@ func TestLoadingSettings(t *testing.T) {
...
+			err := cfg.Load(&CommandLineArgs{
+				HomePath: "../../",
+				Args:     []string{"cfg:auth.proxy.sync_ttl=5"},
+			})
+			So(err, ShouldBeNil)
+
+			So(AuthProxySyncTtl, ShouldEqual, 5)
+		})
+
+		Convey("ldap_sync should not override ldap_sync_ttl that is larger then -1", func() {
```
[the remark this answers is not in the document]
Comment (jongyllen) [reply pkg/setting/setting_test.go]: Thanks, fixed

File: conf/defaults.ini [hunk 4]
```diff
@@ -434,6 +434,7 @@ header_name = X-WEBAUTH-USER
 header_property = username
 auto_sign_up = true
 ldap_sync_ttl = 60
+sync_ttl = 60
```
Comment (torkelo) [inline conf/defaults.ini]: Missed adding this to sample.ini

File: pkg/setting/setting.go (hunk 1 above)
Comment (torkelo) [reply pkg/setting/setting.go]: was this forgotten?

Comment (jongyllen) [reply pkg/setting/setting.go]: Yes, sorry!

**Pull request merged**